main.go

```diff
@@ -426,3 +491,82 @@ func getDefaultGwIf() (string, error) {
 	}
 	return "", fmt.Errorf("not routes found")
 }
+
+func waitForPodCIDR(ctx context.Context, client clientset.Interface, nodeName string) (*v1.Node, error) {
+	// since allocators can assign the podCIDR after the node registers, we do a watch here to wait
+	// for podCIDR to be assigned, instead of assuming that the Get() on startup will have it.
+	ctx, cancelFunc := context.WithTimeout(ctx, 5*time.Minute)
+	defer cancelFunc()
+
+	fieldSelector := fields.OneTermEqualSelector("metadata.name", nodeName).String()
+	lw := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (object runtime.Object, e error) {
+			options.FieldSelector = fieldSelector
+			return client.CoreV1().Nodes().List(ctx, options)
+		},
+		WatchFunc: func(options metav1.ListOptions) (i watch.Interface, e error) {
+			options.FieldSelector = fieldSelector
+			return client.CoreV1().Nodes().Watch(ctx, options)
+		},
+	}
+	condition := func(event watch.Event) (bool, error) {
+		// don't process delete events
+		if event.Type != watch.Modified && event.Type != watch.Added {
+			return false, nil
+		}
+
+		n, ok := event.Object.(*v1.Node)
+		if !ok {
+			return false, fmt.Errorf("event object not of type Node")
+		}
+		// don't consider the node if is going to be deleted and keep waiting
+		if !n.DeletionTimestamp.IsZero() {
+			return false, nil
+		}
+		return n.Spec.PodCIDR != "" && len(n.Spec.PodCIDRs) > 0, nil
+	}
+
+	evt, err := toolswatch.UntilWithSync(ctx, lw, &v1.Node{}, nil, condition)
+	if err != nil {
+		return nil, fmt.Errorf("timeout waiting for PodCIDR allocation: %v", err)
+	}
+	if n, ok := evt.Object.(*v1.Node); ok {
+		return n, nil
+	}
+	return nil, fmt.Errorf("event object not of type node")
+}
+
+// parse pod CIDR to the format that will be useful to BPF code
+func parsePodCIDR(podCIDR string) (podNetBytes []uint32, podMaskBytes []uint32, podMaskSize uint32, err error) {
```

Comment:
I don't like this pattern of encapsulating multiple functionalities in one function and ignore some of the output in one places, one function should just do one thing,  commented in the requirement of the podMaskBytes, so you can just receive here 
```suggestion
func parsePodCIDR(podCIDR *ip.Net) (podNetBytes []uint32, podMaskBytes []uint32, podMaskSize uint32, err error) {
```

Reply:
Deleted this function, see comment above.